File: Data Structures & Algorithms/duplicate-integer/submission-31.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let res = new Set()

        for (let n of nums){
            if(res.has(n)){
                return true
            }

            res.add(n)
        }
        return false
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-31.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // hashmap to store the numbers interated over
        // compare each number to the difference in current nums[i] and target

        Map<Integer, Integer> prevMap = new HashMap<>();

        for(int i = 0; i < nums.length; i++){
            int diff = target - nums[i];
            if(prevMap.containsKey(diff)){
                return new int[] {prevMap.get(diff), i};
            }

            prevMap.put(nums[i], i);
        }

        return new int[] {};
        
    }
}
